python/ray/serve/air_integrations.py

```diff
@@ -184,12 +184,15 @@ def __init__(
         predict_kwargs: Optional[Dict[str, Any]] = None,
         **predictor_from_checkpoint_kwargs,
     ):
-        predictor_cls = _load_predictor_cls(predictor_cls)
-        checkpoint = _load_checkpoint(checkpoint)
+        if isinstance(predictor_from_checkpoint_kwargs.get("user_config"), dict):
+            self.reconfigure(predictor_from_checkpoint_kwargs.get("user_config"))
+        else:
```

Comment:
we don't need this condition because `user_config` should not be part of `predictor_from_checkpoint_kwargs`